src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala

```diff
@@ -217,6 +219,106 @@ class PushdownEnhancement02 extends IntegrationSuiteBase {
     }
   }
 
+  test("test pushdown functions date_add/date_sub") {
+    jdbcUpdate(s"create or replace table $test_table_date " +
+      s"(d1 date)")
+    jdbcUpdate(s"insert into $test_table_date values " +
+      s"('2020-07-28')")
+
+    val tmpDF = sparkSession.read
+      .format(SNOWFLAKE_SOURCE_NAME)
+      .options(thisConnectorOptionsNoTable)
+      .option("dbtable", test_table_date)
+      .load()
+
+    val resultDF = tmpDF.select(
+      col("d1"),
+      date_add(col("d1"),4).as("date_add"),
+      date_sub(col("d1"),4).as("date_sub")
+    )
+
+    val expectedResult = Seq(
+      Row(
+        Date.valueOf("2020-07-28"),
+        Date.valueOf("2020-08-01"),
+        Date.valueOf("2020-07-24"))
+    )
+
+    testPushdown(
+      s"""SELECT (
+         |  "SUBQUERY_0"."D1" ) AS "SUBQUERY_1_COL_0" ,
+         |  ( DATEADD(day, 4 , "SUBQUERY_0"."D1" ) ) AS "SUBQUERY_1_COL_1" ,
+         |  ( DATEADD(day, (0 - ( 4 )), "SUBQUERY_0"."D1" ) ) AS "SUBQUERY_1_COL_2"
+         |FROM (
+         |  SELECT * FROM (
+         |    $test_table_date
+         |  ) AS "SF_CONNECTOR_QUERY_ALIAS"
+         |) AS "SUBQUERY_0"
+         |""".stripMargin,
+      resultDF,
+      expectedResult
+    )
+  }
+
```

Comment:
To reviewer:
This test case is new test when re-adding DateSub.
It tests DateSub/DataAdd to work with the last day on Feb in leap/non-leap year.